pkg/trtl/honu_test.go

```diff
@@ -116,6 +118,99 @@ func (s *trtlTestSuite) TestGet() {
 	require.True(proto.Equal(expectedMeta, reply.Meta))
 }
 
+// Test that we can call the Put RPC and get the correct response.
+func (s *trtlTestSuite) TestPut() {
+	require := s.Require()
+	alice := dbFixtures["alice"]
+	object := dbFixtures["object"]
+	ctx := context.Background()
+
+	// Start the gRPC client.
+	conn, err := s.connect()
+	require.NoError(err)
+	defer conn.Close()
+	client := pb.NewTrtlClient(conn)
+
+	// Put a value to a reserved namespace - should fail.
+	_, err = client.Put(ctx, &pb.PutRequest{
+		Namespace: "default",
+		Key:       []byte(object.Key),
+		Value:     []byte("foo"),
+	})
+	require.Error(err)
+
+	// Put a value without the key - should fail.
+	_, err = client.Put(ctx, &pb.PutRequest{
+		Namespace: object.Namespace,
+		Value:     []byte("foo"),
+	})
+	require.Error(err)
+
+	// Put without a value - should fail.
+	_, err = client.Put(ctx, &pb.PutRequest{
+		Namespace: object.Namespace,
+		Key:       []byte(object.Key),
+	})
+	require.Error(err)
+
+	// Put a value to the default namespace.
+	reply, err := client.Put(ctx, &pb.PutRequest{
+		Key:   []byte("testKey"),
+		Value: []byte("testVal"),
+	})
+	require.NoError(err)
+	require.True(reply.Success)
+	require.Empty((reply.Meta))
+
+	// Put a value to a valid namespace with return_meta=false.
+	reply, err = client.Put(ctx, &pb.PutRequest{
+		Namespace: alice.Namespace,
+		Key:       []byte(alice.Key),
+		Value:     []byte("arlo guthrie"),
+		Options: &pb.Options{
+			ReturnMeta: false,
+		},
+	})
+	require.NoError(err)
+	require.True(reply.Success)
+	require.Empty((reply.Meta))
+
+	// Put a value with return_meta=true.
+	expectedPID, err := strconv.Atoi((os.Getenv("TRTL_REPLICA_PID")))
+	require.NoError(err)
+	expectedRegion := os.Getenv("TRTL_REPLICA_REGION")
+	expectedMeta := &pb.Meta{
+		Key:       []byte(alice.Key),
+		Namespace: alice.Namespace,
+		Region:    metaRegion,
+		Owner:     metaOwner,
+		Version: &pb.Version{
+			Pid:     uint64(expectedPID),
+			Version: 4,
+			Region:  expectedRegion,
+		},
+		Parent: &pb.Version{
+			Pid:     uint64(expectedPID),
+			Version: 3,
+			Region:  expectedRegion,
+		},
+	}
+	reply, err = client.Put(ctx, &pb.PutRequest{
+		Namespace: alice.Namespace,
+		Key:       []byte(alice.Key),
+		Value:     []byte("cheshire cat"),
```

Comment:
I just had one random thought - this request should be changing the state of the underlying database from the JSON in the fixture to this value here, which means a subsequent test that tests for the value might error. The tests are fine now, but I wanted to raise the point in case we encounter it in future, to help us find the problem more quickly.